frontend/src/__tests__/UserList.test.js

```diff
@@ -22,6 +34,256 @@ describe('<UserList />', () => {
+                {
+                  node: {
+                    id: 'ODY0MDEzMTE1NA==',
+                    userName: 'testuser@testemail.gc.ca',
+                    admin: false,
+                    tfa: false,
+                    displayName: 'Test User',
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    ]
+
+    // create a history object and inject it so we can inspect it afterwards
+    // for the side effects of our form submission (a redirect to /!).
+    const history = createMemoryHistory({
+      initialEntries: ['user-list'],
+      initialIndex: 0,
+    })
+
+    // Set the inital history item to user-list
+    const { container, queryAllByRole, getByText } = render(
+      <ThemeProvider theme={theme}>
+        <I18nProvider i18n={i18n}>
+          <Router history={history}>
+            <MockedProvider mocks={mocks} addTypename={false}>
+              <UserList />
+            </MockedProvider>
+          </Router>
+        </I18nProvider>
+      </ThemeProvider>,
+    )
+    expect(container).toBeDefined()
+
+    expect(getByText('Loading...')).toBeInTheDocument()
+    const loadingElement = getByText('Loading...')
+
+    await waitForElementToBeRemoved(loadingElement)
+
+    // Get all of the mocked user cards, and expect there to be only one entry.
+    const userCards = queryAllByRole('userCard')
+    expect(userCards).toHaveLength(1)
+
+    const leftClick = { button: 0 }
+    fireEvent.click(userCards[0], leftClick)
+    // default `button` property for click events is set to `0` which is a left click.
+
+    await waitFor(() => {
+      // Path should be '/user', so expect that value
+      expect(history.location.pathname).toEqual('/user')
+    })
+  })
+
+  it('badges are green when TwoFactor and Admin values are true', async () => {
+    const mocks = [
+      {
+        request: {
+          query: QUERY_USERLIST,
+        },
+        result: {
+          data: {
+            userList: {
+              organization: 'TEST',
+              pageInfo: {
+                hasNextPage: true,
+                hasPreviousPage: true,
+              },
+              edges: [
+                {
+                  node: {
+                    id: 'ODY0MDEzMTE1NA==',
+                    userName: 'testuser@testemail.gc.ca',
+                    admin: true,
+                    tfa: true,
+                    displayName: 'Test User',
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    ]
+
+    const { container, queryAllByRole, getByText } = render(
+      <ThemeProvider theme={theme}>
+        <I18nProvider i18n={i18n}>
+          <MemoryRouter>
+            <MockedProvider mocks={mocks} addTypename={false}>
+              <UserList />
+            </MockedProvider>
+          </MemoryRouter>
+        </I18nProvider>
+      </ThemeProvider>,
+    )
+    expect(container).toBeDefined()
```

Comment:
`container` [will always be defined](https://testing-library.com/docs/react-testing-library/api#container-1) here as it's the `<div>` within which the component is rendered.